### PR TITLE
Image Editor Crop: use local state instead of Redux for better performance

### LINKED ISSUE #8286: Image Editor: Large Image Performance
Let's do a profiling pass on why the image editor chugs along on large images. If you find any bottlenecks, let's try to find work arounds.

### PR DESCRIPTION
**Pull request by lamosty**
Fixes #8286. Instead of storing crop box and canvas bounds in Redux state tree, use local component's state. This increases performance because dispatching actions after moving/resizing the cropbox caused a few unrelated components to update/re-render in the background thus slowing down the swiftness of the Image Editor in general.

The slowness of Image Editor Crop was most noticeable on larger images but it was present in smaller images as well.

## Testing

1. Open Image Editor with a larger image (I used [this image](https://lamostyfree.files.wordpress.com/2016/08/pleiades_large1.jpg));
2. Try moving and resizing the crop box. Just play with it as fast and as much as possible;
3. Is it quick/swift enough?
4. Try comparing it with the previous implementation.

### With Redux

selection_031

### With local state

selection_032

(I know I would have to perform the exact same things for the comparison to be 100% correct but this got me at least somewhat closer. Anyways, the screenshots are just an addition. You can try it yourself. :))

Props @gwwar for telling me to use the Chrome DevTools timeline.

/cc @gwwar @artpi @retrofox @adambbecker

### CODE REVIEW

**Comment by lamosty** [comment]
Note: lot of the changes are just ESLint fixes. Sorry, I couldn't resist. Next time I'll try to do them in another commit for easier reviewing..

File: client/post-editor/media-modal/image-editor/image-editor-crop.jsx [hunk 1]
```diff
@@ -75,13 +65,15 @@ const MediaModalImageEditorCrop = React.createClass( {
...
+		const { crop } = this.state;
+
 		if ( ! isEqual( this.props.bounds, newProps.bounds ) ) {
 			const imageWidth = newProps.bounds.rightBound - newProps.bounds.leftBound;
 			const imageHeight = newProps.bounds.bottomBound - newProps.bounds.topBound;
-			const newTop = newProps.bounds.topBound + newProps.crop.topRatio * imageHeight;
-			const newLeft = newProps.bounds.leftBound + newProps.crop.leftRatio * imageWidth;
-			const newBottom = newTop + newProps.crop.heightRatio * imageHeight;
-			const newRight = newLeft + newProps.crop.widthRatio * imageWidth;
+			const newTop = newProps.bounds.topBound + crop.topRatio * imageHeight;
```
Comment (gwwar) [inline client/post-editor/media-modal/image-editor/image-editor-crop.jsx]: It looks like I'm getting a JS error here for `crop.topRatio` when rotating the image.

File: client/post-editor/media-modal/image-editor/image-editor-crop.jsx [hunk 2]
```diff
@@ -12,13 +12,10 @@ import classNames from 'classnames';
  */
 import Draggable from 'components/draggable';
 import {
-	getImageEditorCropBounds,
 	getImageEditorAspectRatio,
 	getImageEditorTransform,
-	getImageEditorCrop
```
Comment (gwwar) [inline client/post-editor/media-modal/image-editor/image-editor-crop.jsx]: Let's remove the unused redux selectors and actions and related tests.

**Comment by gwwar** [comment]
Great 🔍 work @lamosty! Now we know that dispatching a ton of redux events in a render loop is 🐢. 😓 I must have totally missed that on the original PR. Yay for profiling!  

 In terms of the PR it looks like we have a bug to iron out and some redux tidying before 🚢

File: client/post-editor/media-modal/image-editor/image-editor-crop.jsx (hunk 1 above)
Comment (lamosty) [inline client/post-editor/media-modal/image-editor/image-editor-crop.jsx]: Fixed in c3cce7c, great catch!

File: client/post-editor/media-modal/image-editor/image-editor-crop.jsx (hunk 2 above)
Comment (lamosty) [inline client/post-editor/media-modal/image-editor/image-editor-crop.jsx]: Fixed in #b60ba64. However, now the Reset button doesn't work for crop movement/resizing. 
selection_029. Should we fix this?

Comment (gwwar) [inline client/post-editor/media-modal/image-editor/image-editor-crop.jsx]: Let's tackle that in another PR

**Comment by gwwar** [comment]
Great work! Let's :shipit:

**Comment by lamosty** [comment]
Thanks for the reviews!

**Pull request merged**